crates/interpreter/src/syntax.rs

```diff
@@ -60,10 +60,28 @@ pub struct FuncType<'m> {
     pub results: ResultType<'m>,
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum Share {
```

Comment:
I'm surprised you don't need to gate those too. I guess it's because they are eventually exported. I'm adding lints to catch those hopefully.

Reply:
So for my own understanding, you would expect rustc to complain about an unused enum? But because it's exported it cannot actually be sure about whether it is being used or not?

Reply:
Yes, the lint I'm thinking about is called `unreachable-pub`. It's when an item is `pub` but is actually not reachable from the public API. The `syntax` module is not public and we only re-export a subset of the stuff, so without the `threads` feature, the `Share` enum should be unreachable triggering a warning.